internal/abci/accumulator.go

```diff
@@ -249,9 +249,9 @@ func (app *Accumulator) CheckTx(req abci.RequestCheckTx) (rct abci.ResponseCheck
 	//create a default response
 	ret := abci.ResponseCheckTx{Code: 0, GasWanted: 1, Data: sub.ChainID, Log: "CheckTx"}
 
-	err = app.chain.CheckTx(sub)
+	checkRes := app.chain.CheckTx(sub)
 
-	if err != nil {
+	if !checkRes.IsOK() {
```

Comment:
The error code is not propagated to the caller. `ret.Code` is always set to `protocol.CodeUnknownError`, so the caller won't see the detailed error code.

In general, I think we just want to return `checkRes`. We want to retain the error logging, but I don't see a reason to create and populate a separate `var ret abci.ResponseCheckTx` given that `checkRes` has everything we want, I believe.